### PR TITLE
LA-13928 Permission check scripts for sap hana

### PR DESCRIPTION
**Pull request by akbasniwal-lb**
sample response stats:

===============Database list with size============
SCHEMA_NAME,DATABASE_SIZE_MB
"KAIFTEST",0.670000
"AUTOMATION_STUFF",0.230000
"BROKER_PO_USER",0.000000

===============Total tables, columns and size per schema============
SCHEMA_NAME,TABLE_COUNT,COLUMN_COUNT
"KAIFTEST",1,4
"AUTOMATION_STUFF",1,12

===============Data type distribution============
DATA_TYPE,COUNT
"NVARCHAR",16

### CODE REVIEW

**Review by arpit-lb — APPROVED** [review]

File: sql_user_check_sap_hana/complete_metadata.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        SYS.INDEX_COLUMNS ic) as cc
+    ON tc.TABLE_NAME = cc.TABLE_NAME
+    AND tc.COLUMN_NAME = cc.COLUMN_NAME
+    AND tc.SCHEMA_NAME = cc.SCHEMA_NAME
+LEFT JOIN
+    SYS.M_TABLES mt
+    ON tc.TABLE_NAME = mt.TABLE_NAME
+    AND tc.SCHEMA_NAME = mt.SCHEMA_NAME
+WHERE
+    tc.SCHEMA_NAME NOT LIKE '_SYS%' AND tc.SCHEMA_NAME NOT LIKE 'SYS%' AND tc.SCHEMA_NAME NOT LIKE 'PAL%' AND tc.SCHEMA_NAME NOT LIKE 'SAP%'
```
[the remark this answers is not in the document]
Comment (kaif-lb) [reply sql_user_check_sap_hana/complete_metadata.sql:31]: can we use schema name as template here and render it by taking schema name as input to the script.

File: sql_user_check_sap_hana/database_stats.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        u) username=${OPTARG};;
+        m) mode=${OPTARG};;
+        o) outputfile=${OPTARG};;
+        p) port=${OPTARG};;
+    esac
+done
+
+echo "host: $host username: $username mode: $mode outputfile: $outputfile port: $port";
+
+# Check if mysql command is available
```
[the remark this answers is not in the document]
Comment (kaif-lb) [reply sql_user_check_sap_hana/database_stats.sh:21]: This isn;t addressed, replace mysql command with hdbsql command

File: sql_user_check_sap_hana/complete_metadata.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaif-lb) [reply sql_user_check_sap_hana/complete_metadata.sql:31]: or if this satisfies all excluded schema then lets continue with it.

File: sql_user_check_sap_hana/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+Please run each of the .sql files on that database with the account credentials that you need to connect to LightBeam.
+
+Prerequisites:
+- Install [hdbsql](https://tools.hana.ondemand.com/#hanatools) on the machine.[Instruction](https://developers.sap.com/tutorials/hana-clients-install..html)
+
+Fetch Schema List:
+```shell
+hdbsql -n <DATABASE_HOST_IP>:<PORT> -u <DATABASE_USER> -p <PASSWORD> -d <DATABASE_NAME> "SELECT SCHEMA_NAME FROM SCHEMAS" > schema_list
```
Comment (kaif-lb) [inline sql_user_check_sap_hana/README.md:8]: shouldn't this be database_stats.sh instead of hdbsql ??

**Review by kaif-lb — APPROVED** [review]

File: sql_user_check_sap_hana/README.md (hunk 3 above)
Comment (akbasniwal-lb) [reply sql_user_check_sap_hana/README.md:8]: this is command for testing if hdbsql is installed and working

**Pull request merged**